pkg/util/kubernetes/apiserver/resourcetypes/mock/discoveryclient.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
```

Comment:
nit: if this is only used for tests i think you can add a test build flag
```suggestion
//go:build kubeapiserver && test
```